src/utils.ts

```diff
@@ -1,5 +1,8 @@
+import { workspace, TextEditor, TextDocumentContentChangeEvent, Position } from "vscode";
+
+import { EXT_NAME } from "./extension";
```

Comment:
This may cause a circular reference issues. I'd suggest to move EXT_NAME/EXT_ID to utils